Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -124,10 +126,28 @@ def tests
           parallel = false
           ["test/#{test_name}_spec.rb:#{line}"]
         end
+      elsif args.changed?
+        changed_files = Utils.safe_popen_read("git", "diff", "--name-only", "master")
+
+        filestub_regex = %r{Library/Homebrew/([A-Za-z/_-]+).rb}
+
+        changed_files.scan(filestub_regex)
+                     .map { |filestub| "test/#{filestub.last}_spec.rb" }
+                     .select { |filename| File.exist?(filename) }
       else
         Dir.glob("test/**/*_spec.rb")
       end
 
+      if files.empty?
+        raise UsageError, "The --only= argument requires a valid file or folder name!" if args.only
+
+        if args.changed?
+          opoo "Either no files have been changed from the master branch " \
+               "or no tests are associated with the changed files!"
+          return
+        end
```

Comment:
```suggestion
        if args.only
          raise UsageError, "The --only= argument requires a valid file or folder name!"
        elsif args.changed?
          opoo "Either no files have been changed from the master branch " \
               "or no tests are associated with the changed files!"
          return
        end
```

Also: does the latter want to maybe be a `UsageError` too? Wondering if any other cases need handled, too.

Reply:
I decided to split those two up into a usage error and a warning.